db/view/view_update_generator.cc

```diff
@@ -40,14 +40,15 @@ future<> view_update_generator::start() {
                 _pending_sstables.wait().get();
             }
 
-            // To ensure we don't race with updates, move the entire content
-            // into a local variable.
-            auto sstables_with_tables = std::exchange(_sstables_with_tables, {});
-
-            // If we got here, we will process all tables we know about so far eventually so there
-            // is no starvation
-            for (auto table_it = sstables_with_tables.begin(); table_it != sstables_with_tables.end(); table_it = sstables_with_tables.erase(table_it)) {
-                auto& [t, sstables] = *table_it;
+            // We process a small batch from each table in turn to ensure fair incremental progress.
+            for (auto table_it = _sstables_with_tables.begin(); table_it != _sstables_with_tables.end();) {
+                auto& [t, all_sstables] = *table_it;
+                std::vector<sstables::shared_sstable> excluded_sstables(all_sstables.begin(), all_sstables.end());
+                std::vector<sstables::shared_sstable> sstables;
+                while (!all_sstables.empty() && sstables.size() < sstable_batch_size) {
```

Comment:
maybe we should sort by first token to preserve token contiguity? we also gain the benefit of overlapping files being processed in the same batch.

Reply:
Yes, makes sense.